File: VB/WebSite/App_Code/ImageHandler.ashx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web
Imports System.Drawing
Imports System.Drawing.Imaging

Public Class ImageHandler
	Implements IHttpHandler
	Public Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
		Dim imgId As String = context.Request("imgId")
		Dim path As String = context.Server.MapPath(String.Format("~/App_Data/{0}.png", imgId))

		context.Response.ContentType = "image/png"

		If System.IO.File.Exists(path) Then
			context.Response.WriteFile(path)
		End If

		context.Response.End()
	End Sub

	Public ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
		Get
			Return False
		End Get
	End Property
End Class

File: CS/WebSite/App_Code/ImageHandler.ashx.cs
using System;
using System.Web;
using System.Drawing;
using System.Drawing.Imaging;

public class ImageHandler : IHttpHandler {
    public void ProcessRequest (HttpContext context) {
        string imgId = context.Request["imgId"];
        string path = context.Server.MapPath(string.Format("~/App_Data/{0}.png", imgId));

        context.Response.ContentType = "image/png";

        if (System.IO.File.Exists(path))
            context.Response.WriteFile(path);
        
        context.Response.End();
    }
 
    public bool IsReusable {
        get {
            return false;
        }
    }
}